packages/addresses/src/base58.ts

```diff
@@ -30,6 +30,13 @@ export function assertIsBase58EncodedAddress(
     }
 }
 
+export function address<TAddress extends string = string>(
+    putativeBase58EncodedAddress: TAddress
+): Base58EncodedAddress<TAddress> {
+    assertIsBase58EncodedAddress(putativeBase58EncodedAddress);
```

Comment:
This is obviously the right thing to do, semantically, I just wish that doing this didn't have a CPU cost.

```ts
// I get this for free.
const address = '555555555555555555555555` as Base58EncodedAddress<'555555555555555555555555'>;

// This calls `base58.serialize` to perform the assertion, ffs
const address = address('555555555555555555555555');
```

Reply:
Doesn't seem like we can avoid it, right? If someone uses the former, they get the CPU bonus but no assertion, and if they instead use the latter, it's the opposite.

Ideally the real slick devs could just _make sure_ their string is a valid base58 address and go with the former

Reply:
Yeah. I'm just whining out loud.